compiler/p/env/OMRCPU.cpp

```diff
@@ -20,15 +20,122 @@
  *******************************************************************************/
 
 #include "env/CPU.hpp"
-
+#include "env/CompilerEnv.hpp"
 #include "env/Processors.hpp"
 #include "infra/Assert.hpp"
 
+bool
+OMR::Power::CPU::getPPCSupportsVMX()
+   {
+   if (TR::Compiler->omrPortLib == NULL)
+      return false;
+
+   return self()->supportsFeature(OMR_FEATURE_PPC_HAS_ALTIVEC);
+   }
+
+bool
+OMR::Power::CPU::getPPCSupportsVSX()
+   {
+   if (TR::Compiler->omrPortLib == NULL)
+      return false;
+
+   return self()->supportsFeature(OMR_FEATURE_PPC_HAS_VSX);
+   }
+
+bool
+OMR::Power::CPU::getPPCSupportsAES()
+   {
+   if (TR::Compiler->omrPortLib == NULL)
+      return false;
+
+   return self()->supportsFeature(OMR_FEATURE_PPC_HAS_ALTIVEC) && self()->isAtLeast(OMR_PROCESSOR_PPC_P8) && self()->supportsFeature(OMR_FEATURE_PPC_HAS_VSX);
+   }
+
+bool
+OMR::Power::CPU::getPPCSupportsTM()
+   {
+   if (TR::Compiler->omrPortLib == NULL)
+      return false;
+
+   return self()->supportsFeature(OMR_FEATURE_PPC_HTM);
+   }
+
+bool
+OMR::Power::CPU::hasPopulationCountInstruction()
+   {
+   if (TR::Compiler->omrPortLib == NULL)
+      return false;
+
+#if defined(J9OS_I5)
+   return ans;
+#endif
+
+   return self()->isAtLeast(OMR_PROCESSOR_PPC_P7);
+   }
+
+bool
+OMR::Power::CPU::supportsDecimalFloatingPoint()
+   {
+   if (TR::Compiler->omrPortLib == NULL)
+      return false;
+   
+   return self()->supportsFeature(OMR_FEATURE_PPC_HAS_DFP);
+   }
+
 bool
 OMR::Power::CPU::getPPCis64bit()
    {
-   TR_ASSERT(self()->id() >= TR_FirstPPCProcessor && self()->id() <= TR_LastPPCProcessor, "Not a valid PPC Processor Type");
-   return (self()->id() >= TR_FirstPPC64BitProcessor)? true : false;
+   if (TR::Compiler->omrPortLib == NULL)
+      {
+      TR_ASSERT(self()->id() >= TR_FirstPPCProcessor && self()->id() <= TR_LastPPCProcessor, "Not a valid PPC Processor Type");
+      return self()->id() >= TR_FirstPPC64BitProcessor;
+      }
+
+   TR_ASSERT(self()->isAtLeast(OMR_PROCESSOR_PPC_FIRST) && self()->isAtMost(OMR_PROCESSOR_PPC_LAST), "Not a valid PPC Processor Type");
+   TR_ASSERT_FATAL((self()->id() >= TR_FirstPPCHwCopySignProcessor) == (self()->isAtLeast(OMR_PROCESSOR_PPC_HW_COPY_SIGN_FIRST)), "getPPCis64bit");
```

Comment:
What is this testing HW_COPY_SIGN for 64bit HW query?
Synchronizing the enum value between TR_yyyProcessor and OMR_PROCESSOR_YYY?  if that is the intention, why not directly testing if they are equal?

Reply:
Typo, fixed